pymc/model.py

```diff
@@ -2035,10 +2035,49 @@ def Deterministic(name, var, model=None, dims=None):
 def Potential(name, var, model=None):
     """Add an arbitrary factor potential to the model likelihood
 
+      The Potential function is used to add arbitrary factors (such as constraints or other likelihood components) to adjust the probability density of the model.
+
+      Have a look at the following example:
+
+     .. code:: python
+
+        with pm.Model() as model:
+          x = pm.Normal('x', mu=0, sigma=1)
+          y = pm.Normal('y', mu=x, sigma=1, observed=data)
+          constraint = x>= 0
+          potential = pm.Potential('x_constraint', -pm.math.log(pm.math.switch(constraint, 1, 1e-30))
+
+        In this example, we define a constraint on x to be greater or equal to 0 via the ``pm.Potential`` function.
+        We pass ``-pm.math.log(pm.math.switch(constraint, 1, 1e-30))`` as second argument which will return an expression depending on if the constraint is met or not and which will be added to the likelihood of the model.
+        The reciprocal exponential function of the negative logarithm is proportional to the likelihood function. When the constraint is met,the switch function returns 1 making the negative logarithm 0 and hence likelihood function has no penalizing effect. When the constraint is not met the switch function returns 1e-30 and the negative logarithm takes a high value(i.e. 30) reciprocal exponential function of which makes the likelihood function very small and successfully penalizing it.
+        The probablity density that this model produces agrees with the constraint that x should be greater than or equal to 0.
+
+     .. code:: python
+
+        with pm.Model():
+          # p(max_items) = 1 / max_items
+          max_items = pm.Uniform("max_items", lower=1, upper=100)
+          pm.Potential('power_prior', -pm.math.log(max_items))
```

Comment:
They are equivalent, but maybe this one is more readable for users

```suggestion
          pm.Potential('power_prior', pm.math.log(1/max_items))
```